vector-im/element-android
# Disable 'Enable biometrics' option if there are not biometric authenticators enrolled.

## Type of change


## Content

Fixes #6713 .

## Motivation and context

Enabling biometric auth in the app is not restricted to users who have biometric authenticators enrolled in the device. This caused `UserNotAuthenticatedException` crashes for users who enabled this option without having those authenticators.

## Screenshots / GIFs





## Tests



- Make sure you have no biometric authenticators enrolled in your device (no fingerprint, no face unlock, etc.).
- Open the app, go to Settings > Security & Privacy > Protect Access.
- Enable PIN code.
- Check 'Enable biometrics' can't be toggled because it's disabled.
- Now add a fingerprint to your device.
- Go back to the app, in 'Protect Access'.
- Check that 'Enable biometrics' can be accessed now.

Note: I had some issues while testing this on emulators, it seems like an emulator bug. Having several fingerprints enrolled and enabling biometrics in the app caused the same `UserNotAuthenticatedException` crash mentioned above. Apparently, `BiometricManager` detects the valid authenticators but `BiometricPrompt` doesn't.

## Tested devices

- OS version(s): 12

## Checklist
